src/backend/joanie/core/factories.py

```diff
@@ -677,6 +682,208 @@ def main_invoice(self, create, extracted, **kwargs):
         return None
 
 
+class OrderGeneratorFactory(factory.django.DjangoModelFactory):
+    """A factory to create an Order"""
+
+    class Meta:
+        model = models.Order
+
+    product = factory.SubFactory(ProductFactory)
+    course = factory.LazyAttribute(lambda o: o.product.courses.order_by("?").first())
+    total = factory.LazyAttribute(lambda o: o.product.price)
+    enrollment = None
+    state = enums.ORDER_STATE_DRAFT
+
+    @factory.lazy_attribute
+    def owner(self):
+        """Retrieve the user from the enrollment when available or create a new one."""
+        if self.enrollment:
+            return self.enrollment.user
+        return UserFactory()
+
+    @factory.lazy_attribute
+    def organization(self):
+        """Retrieve the organization from the product/course relation."""
+        if self.state == enums.ORDER_STATE_DRAFT:
+            return None
+
+        course_relations = self.product.course_relations
+        if self.course:
+            course_relations = course_relations.filter(course=self.course)
+        return course_relations.first().organizations.order_by("?").first()
+
+    @factory.post_generation
+    def main_invoice(self, create, extracted, **kwargs):
+        """
+        Generate invoice if needed
+        """
+        if create:
+            if extracted is not None:
+                # If a main_invoice is passed, link it to the order.
+                extracted.order = self
+                extracted.save()
+                return extracted
+
+            from joanie.payment.factories import (  # pylint: disable=import-outside-toplevel, cyclic-import
+                InvoiceFactory,
+            )
+
+            return InvoiceFactory(order=self)
```

Comment:
I think you should also set the `total` field of the invoice to match the order total value.